geometric_kernels/_logging.py

```diff
@@ -1,5 +1,58 @@
 """ Setup logging """
 import logging
 
-logging.basicConfig(format="%(levelname)s: %(message)s")
-logging.getLogger().setLevel(logging.INFO)
+
+class DisableLogging:
+    """
+    Adapted from https://stackoverflow.com/a/20251235. Use as
+    ```python
+    with DisableLogging():
+        do your stuff
+    ```
+    """
+
+    def __enter__(self):
+        logging.disable(logging.CRITICAL)
+
+    def __exit__(self, exit_type, exit_value, exit_traceback):
+        logging.disable(logging.NOTSET)
+
+
+class FirstPartFilter(logging.Filter):
+    """
+    A filter that provides the `name_first` variable for formatting. For a logger
+    called "aaa.bbb.ccc", name_first="aaa".
+    Adapted from https://stackoverflow.com/a/46961676.
+    """
+
+    def filter(self, record):
+        record.name_first = record.name.rsplit(".", 1)[0]
+        return True
+
+
+class NoUsingBackendFilter(logging.Filter):
+    """
+    A filter that removes the "Using *** backend" log record of geomstats.
+    """
+
+    def filter(self, record):
+        msg = record.getMessage()
+        # TODO: when geomstats implements better logging, add
+        # msg.name_first == "geomstats"
+        # as the third condition for filtering.
+        return not (msg.startswith("Using ") and msg.endswith(" backend"))
+
+
+root_handler = logging.StreamHandler()
+root_handler.addFilter(FirstPartFilter())
+root_handler.addFilter(NoUsingBackendFilter())
+formatter = logging.Formatter("%(levelname)s (%(name_first)s): %(message)s")
+root_handler.setFormatter(formatter)
+# Note: using baseConfig allows the "outermost" code to define the logging
```

Comment:
While this is cool, I think this might be problematic in the long run. First of, what makes you think that our library code is outermost, or that our library code is to dictate this policy for the subsequent code? Imagine using this library in some script, and the user sets some logging config only to find out that it does not work, then wasting hours of debugging to find the root cause being this little line here. 

Another this is unequivocally disabling the geomstats warning. One can imagine some people relying on that in their scripts.

I think using this is fine for _our own_ outermost code (like for example the jupyter notebooks -- we can also add some comments there about the whats and the whys) but I'm hesitant about putting that into the library-wide `__init__`